distributed/client.py

```diff
@@ -3037,10 +3037,18 @@ def _expand_key(cls, k):
         or retries dictionary.
         """
         if not isinstance(k, tuple):
+            if not dask.is_dask_collection(k):
+                raise TypeError("key should be a tuple of dask collection")
```

Comment:
This isn't true.  Users are also allowed to put actual keys here as well.

Reply:
Ah, ok.
By my reading, any kind of object can be input and it will be wrapped in a tuple. However, if that object can't be used to make a proper key, the requirement will be silently ignored.
Is there a definition of a proper key?